annotated.go

```diff
@@ -461,21 +478,22 @@ func (ann *annotated) results() (
 				}
 				continue
 			}
-
-			out.Field(offsets[i]).Set(r)
+			for j := range resTypes {
+				if fieldIdx := offsets[j][i]; fieldIdx > 0 {
+					outResults[j].Field(fieldIdx).Set(r)
```

Comment:
consider adding a comment here about why `fieldIdx` has to be non-zero to be valid